src/coreclr/jit/morphblock.cpp

```diff
@@ -1482,19 +1502,32 @@ GenTree* MorphCopyBlockHelper::CopyFieldByField()
 
         if (addrSpillAsg != nullptr)
         {
-            asgFields    = m_comp->gtNewOperNode(GT_COMMA, TYP_VOID, addrSpillAsg, asgOneFld)->AsOp();
+            result       = m_comp->gtNewOperNode(GT_COMMA, TYP_VOID, addrSpillAsg, asgOneFld)->AsOp();
             addrSpillAsg = nullptr;
         }
-        else if (asgFields != nullptr)
+        else if (result != nullptr)
+        {
+            result = m_comp->gtNewOperNode(GT_COMMA, TYP_VOID, result, asgOneFld)->AsOp();
+        }
+        else
+        {
+            result = asgOneFld;
+        }
+    }
+
+    if (dyingFieldCnt == fieldCnt)
```

Comment:
It is a little non-obvious why does this check reside here instead of before the loop.

Reply:
Moved it up.